src/lib/set.js

```diff
@@ -14,7 +18,14 @@ module.exports = (obj, bigKey, val) => {
     if (i === keys.length - 1) {
       curr[key] = val;
     } else {
-      if (!Object.prototype.hasOwnProperty.call(curr, key)) {
+      if (
+        !Object.prototype.hasOwnProperty.call(curr, key) ||
+        isBoolean(curr[key]) ||
+        isNull(curr[key]) ||
+        isUndefined(curr[key]) ||
+        isNumber(curr[key]) ||
+        isString(curr[key])
+      ) {
```

Comment:
I did this a little while ago, and now I'm not so sure this is ... completely thought out. 

I lose information this way. For example, imagine this case:
```
const obj = {a: true};

set(obj, 'a.b', true);
```

I lose the fact that `a === true`. I'm not sure how much this matters, because I wouldn't do something like this on purpose, but I should probably convert a `primitive boolean` like `true` to a `Boolean object` like `new Boolean(true)`.

Reply:
Most primitives should be promoted now.